examples/platform_channel/macos/Runner/MainFlutterWindow.swift

```diff
@@ -0,0 +1,84 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Cocoa
+import FlutterMacOS
+
+class MainFlutterWindow: NSWindow, FlutterStreamHandler, PowerSourceStateChangeDelegate {
+  private let powerSource = PowerSource()
+  private let stateChangeSource = PowerSourceStateChangeHandler()
+  private var eventSink: FlutterEventSink?
+
+  override func awakeFromNib() {
+    let flutterViewController = FlutterViewController.init()
+    let windowFrame = self.frame
+    self.contentViewController = flutterViewController
+    self.setFrame(windowFrame, display: true)
+
+    // Register battery method channel.
+    let registrar = flutterViewController.registrar(forPlugin: "BatteryLevel")
+    let batteryChannel = FlutterMethodChannel(
+      name: "samples.flutter.io/battery",
+      binaryMessenger: registrar.messenger)
+    batteryChannel.setMethodCallHandler({ [weak self] (call, result) in
```

Comment:
Changed it to simply capture `powerSource` retained. The retain cycle is fine since none of these objects are ever collected. Closing the window shuts down the app.